keps/sig-storage/20180515-svcacct-token-volumes.md

```diff
@@ -159,6 +161,55 @@ sources:
     audience: ca.istio.io
 ```
 
+### File Permission
+
+The secret projections are currently written with world readable (0644) file
+permissions. Given that file permissions are one of the oldest and most hardened
+isolation mechanisms on unix, this is not ideal. We would like to
+opportunistically restrict permissions for projected service account tokens as
+long we can show that they won’t break users if we are to migrate away from
+secrets to distribute service account credentials.
+
+#### Proposed heuristics
+
++   *Case 1*: The pod has an fsGroup set. We can set the file permission on the
+    token file to 0600 and let the fsGroup mechanism work as designed. It will
+    set the permissions to 0640, chown the token file to the fsGroup and start
+    the containers with a supplemental group that grants them access to the
+    token file. This works today.
++   *Case 2*: The pod’s containers declare the same runAsUser for all containers
```

Comment:
and if it's not the same uid across containers, then we go to world readable fallback? are we aware of any mutating admission plugins that would cause the fallback to always happen?

Reply:
i am not aware of a mutating admission plugin on runAsUser. the most relevant one is PSP admission controller which is a validating plugin.